common/persistence/persistenceInterface.go

```diff
@@ -273,6 +272,44 @@ type (
 		TaskQueueInfo *commonpb.DataBlob
 	}
 
+	InternalUpdateTaskQueueRequest struct {
+		NamespaceID   string
+		TaskQueue     string
+		TaskType      enumspb.TaskQueueType
+		TaskQueueKind enumspb.TaskQueueKind
+		RangeID       int64
+		ExpiryTime    *time.Time
+		TaskQueueInfo *commonpb.DataBlob
+	}
+
+	InternalCreateTasksRequest struct {
+		NamespaceID   string
+		TaskQueue     string
+		TaskType      enumspb.TaskQueueType
+		RangeID       int64
+		TaskQueueInfo *commonpb.DataBlob
+		Tasks         []*InternalCreateTask
+	}
+
+	InternalCreateTask struct {
+		TaskId     int64
+		ExpiryTime *time.Time
+		Task       *commonpb.DataBlob
```

Comment:
worth converting this to a non pointer struct? i.e. `Task       *commonpb.DataBlob` vs `Task       commonpb.DataBlob`

Reply:
same applies to other slices

Reply:
The NewDataBlob() returns *commonpb.DataBlob. All the deserialization methods takes *commonpb.DataBlob, and all serialization returns *commonpb.DataBlob. Use struct would work, but then all these places with have to use additional * or &. Will leave it as is in tis PR. I think it is worth to have more discussion and maybe change those serialization / deserialization methods to take struct instead of pointer.